### DISCUSSION TITLE
How can I realize data saving?

### DISCUSSION THREAD

**Discussion by Creepy0964** [CosmosOS/Cosmos#1847] [Q&A]
I want to save some data from OS and when PC reboots, it will load saved data, but how can I make this?

**Comment by valentinbreiz** [comment]
Hello @m4dc1t4, you can save your data in files using the filesystem. I let you read our article about the Cosmos FileSystem implementation: https://cosmosos.github.io/articles/Kernel/VFS.html

**Reply by Creepy0964** [reply]
Thank you! But can I write variables somewhere in such files?
